pkg/engine/renderer/dependencies.go

```diff
@@ -190,6 +222,9 @@ func (de *dependencyCalculator) resourceDependency(d resourceDependency) (*unstr
 	}
 
 	err := de.Client.Get(context.TODO(), key, dep)
+	if apierrors.IsNotFound(err) {
```

Comment:
this is semantics or opinion (so not blocking to me) but I would rather have the `isNotFound(err)` by the calling client than pass a (nil, nil) as something meaningful

Reply:
I've decided against it. I've tried it, and i'd have to either:

`return err` here, without adding an `fmt.Errorf("some nice message %w", err)`. At least it's a non-public func.

or I have to do 

`if (apierrors.IsNotFound(errors.Unwrap(err)) {` on the caller side.

Both solutions don't really sit well with me.

I get where you're coming from, but in this case I prefer the `nil, nil` return.